--- a/package/chrony/chrony.mk
+++ b/package/chrony/chrony.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-CHRONY_VERSION = 4.8
+CHRONY_VERSION = 4.9
 CHRONY_SITE = https://chrony-project.org/releases
 CHRONY_LICENSE = GPL-2.0
 CHRONY_LICENSE_FILES = COPYING
--- a/package/chrony/chrony.hash
+++ b/package/chrony/chrony.hash
@@ -1,5 +1,3 @@
-# Locally calculated after checking pgp signature
-# https://chrony-project.org/releases/chrony-4.8-tar-gz-asc.txt
-sha256  33ea8eb2a4daeaa506e8fcafd5d6d89027ed6f2f0609645c6f149b560d301706  chrony-4.8.tar.gz
-# Locally calculated
+# Locally calculated from the official HTTPS chrony 4.9 release archive.
+sha256  4924c6f530105bcd5b9e9e33c48a2ae1bfd889222c8480bc41601110efc864d0  chrony-4.9.tar.gz
 sha256  ab15fd526bd8dd18a9e77ebc139656bf4d33e97fc7238cd11bf60e2b9b8666c6  COPYING
